modules/hal_silabs/simplicity_sdk/CMakeLists.txt

```diff
@@ -23,8 +23,15 @@ set(BLOBS_DIR      ${ZEPHYR_HAL_SILABS_MODULE_DIR}/zephyr/blobs/simplicity_sdk)
 # respectively.
 string(TOUPPER ${CONFIG_SOC_SERIES} SILABS_DEVICE_FAMILY)
 
-# Get SoC series number, i.e. translate e.g. efr32bg22 -> 22
-string(SUBSTRING ${CONFIG_SOC_SERIES} 7 2 SILABS_DEVICE_FAMILY_NUMBER)
+if("${CONFIG_SOC_PART_NUMBER}" MATCHES "MGM24")
+  set(SILABS_DEVICE_IS_MODULE TRUE)
+  set(SILABS_DEVICE_FAMILY "MGM24")
```

Comment:
This option and the two below should all be derived from SOC_* Kconfig options, not hard-coded in CMake.